### PR TITLE
Add derivative for 'padded(forSizes:with:)'.

### LINKED ISSUE #179: Define a derivative for 'padded(forSizes:with:)'.
A custom derivative should be defined for API [`padded(forSizes:with:)`](https://github.com/tensorflow/swift-apis/blob/55adef71ec36c9ee660ab590179692863a247172/Sources/TensorFlow/Operators/Basic.swift#L677).

### PR DESCRIPTION
**Pull request by Dash2507**
Resolves #179.

### CODE REVIEW

**Comment by Dash2507** [comment]
Fix: #179

**Comment by Dash2507** [comment]
@eaplatanios All the requested changes are done.

File: Sources/TensorFlow/Operators/Basic.swift [hunk 1]
```diff
@@ -682,6 +683,26 @@ public extension Tensor where Scalar: Numeric {
...
+            forSizes sizes: [(before: Int, after: Int)],
+            with value: Scalar = 0
+    ) -> (Tensor, (Tensor) -> Tensor) {
+        let result = padded(forSizes: sizes, with: value)
+        return (result, { [rank = rankTensor, shape = shapeTensor] v in
+            let paddings = Tensor<Int32>(
+                    shape: [sizes.count, 2],
+                    scalars: sizes.flatMap { [Int32($0.before), Int32($0.after)] })
+            let padBefore = Raw.slice(paddings,
+                    begin: Tensor<Int32>([0, 0]),
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/Basic.swift]:
```suggestion
                begin: Tensor<Int32>([0, 0]),
```

File: Sources/TensorFlow/Operators/Basic.swift [hunk 2]
```diff
@@ -682,6 +683,26 @@ public extension Tensor where Scalar: Numeric {
...
+                    size: Tensor<Int32>(stacking: [rank, Tensor<Int32>(1)]))
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/Basic.swift]:
```suggestion
                size: Tensor<Int32>(stacking: [rank, Tensor<Int32>(1)]))
```

File: Sources/TensorFlow/Operators/Basic.swift [hunk 3]
```diff
@@ -682,6 +683,26 @@ public extension Tensor where Scalar: Numeric {
...
+internal extension Tensor where Scalar: TensorFlowFloatingPoint {
+    @inlinable
+    func _vjpPadded(
+            forSizes sizes: [(before: Int, after: Int)],
+            with value: Scalar = 0
+    ) -> (Tensor, (Tensor) -> Tensor) {
+        let result = padded(forSizes: sizes, with: value)
+        return (result, { [rank = rankTensor, shape = shapeTensor] v in
+            let paddings = Tensor<Int32>(
+                    shape: [sizes.count, 2],
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/Basic.swift]:
```suggestion
                shape: [sizes.count, 2],
```

File: Sources/TensorFlow/Operators/Basic.swift [hunk 4]
```diff
@@ -682,6 +683,26 @@ public extension Tensor where Scalar: Numeric {
...
+                    scalars: sizes.flatMap { [Int32($0.before), Int32($0.after)] })
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/Basic.swift]:
```suggestion
                scalars: sizes.flatMap { [Int32($0.before), Int32($0.after)] })
```

File: Tests/TensorFlowTests/OperatorTests/BasicTests.swift [hunk 5]
```diff
@@ -32,6 +32,23 @@ final class BasicOperatorTests: XCTestCase {
...
+        let x = Tensor<Float>(ones: [2, 2])
+        let target = Tensor<Float>([[3, 3, 3], [1, 1, 3], [1, 1, 3]])
+        let paddedTensor = x.padded(forSizes: [(1, 0), (0, 1)], with: 3.0)
+        XCTAssertEqual(paddedTensor, target)
+    }
+
+    func testVJPPadded() {
+        let x = Tensor<Float>(ones: [3, 2])
+        let target = Tensor<Float>([[2, 2], [2, 2], [2, 2]])
+        let grads = x.gradient{ a -> Tensor<Float> in
```
Comment (rxwei) [inline Tests/TensorFlowTests/OperatorTests/BasicTests.swift]:
```suggestion
        let grads = x.gradient { a -> Tensor<Float> in
```

**Pull request merged**